pkg/monitoring/chain_reader.go

```diff
@@ -19,69 +23,54 @@ type ChainReader interface {
 // NewChainReader produces a ChainReader that issues requests to the Terra RPC
 // in sequence, even if it's called by multiple sources in parallel.
 // That's because the Terra endpoint is aggresively rate limitting the monitor.
-func NewChainReader(client *pkgClient.Client) ChainReader {
+func NewChainReader(terraConfig TerraConfig, coreLog logger.Logger) ChainReader {
 	return &chainReader{
-		client,
-		sync.Mutex{},
+		terraConfig,
+		coreLog,
 		sync.Mutex{},
+		ratelimit.New(1,
+			ratelimit.Per(5*time.Second), // one request every 5 seconds
```

Comment:
spoke offline - would be good to make an env var so we can quickly tune if needed